### DISCUSSION TITLE
How to add country names to Locales?

### DISCUSSION THREAD

**Discussion by amityweb** [vercel/next.js#46517] [Help]
module.exports is the following for example:
```
i18n: {
	locales: ['en', 'fr-FR'],
	defaultLocale: 'en',
	localeDetection: false,
},
```
We add a menu based on locales as follows:
```
const { locale, locales, asPath } = useRouter();
{locales.map((l, i) => {
	return (
		<span key={i} className={l === locale ? 'selected' : ''}>
			<ul>
				<li><Link href={asPath} locale={l}>{l}</Link>  |  </li>
			</ul>
		</span>
	);
})}
```

The menu ends up being:
`en  |  fr-FR`

Thats obviously not a good menu. We would like 
`US | France`

How can we add in the actual country names and access them in the menu loop? 

We need to avoid coding in the names as the locales are pulled in from our CMS, so they can be added dynamically. 

Thanks

**Comment by ciruz** [comment]
Hey,

checkout [Intl.DisplayNames](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DisplayNames.).

You can output translated region names:

```javascript
const regionNamesInEnglish = new Intl.DisplayNames(['en'], { type: 'region' }); // In Englisch
console.log(regionNamesInEnglish.of('US'));
// Output: United States

console.log(regionNamesInEnglish.of('AT'));
// Output: Austria

const regionNamesInEnglish = new Intl.DisplayNames(['de'], { type: 'region' }); // In German
console.log(regionNamesInEnglish.of('AT'));
// Output: Österreich
```

This also works for the language name itself, just switch the type to ``language``, like so:

```javascript
const languageNamesInEnglish = new Intl.DisplayNames(['en'], { type: 'language' });
console.log(languageNamesInEnglish.of('de'))
// Output: German

languageNamesInEnglish.of('en')
// Output: English

languageNamesInEnglish.of('de-at')
// Output: Austrian German
```

**Comment by amityweb** [comment]
Oh not seen that before, that's useful. 

But can it do countries? We would like our menu to be country based, so for fr-FR we would output France as the menu link. Because our sites are for country based websites, not just language. 

Thanks

**Reply by amityweb** [reply]
Oh I can see I can add { type: 'region' }, so will try that, although not quite working right just yet.

**Reply by ciruz** [reply]
Sure the region names, but initialize it with the correct language if you want every region in their language.

```JavaScript
const regionNamesInEnglish = new Intl.DisplayNames(['en'], { type: 'region' }); // In Englisch
console.log(regionNamesInEnglish.of('US'));
// Output: United States
const regionNamesInGerman = new Intl.DisplayNames(['de'], { type: 'region' }); // In German
console.log(regionNamesInGerman.of('DE'));
// Output: Deutschland 

```
